src/main/java/eu/bbmri_eric/negotiator/service/NegotiationServiceImpl.java

```diff
@@ -307,21 +309,38 @@ public NegotiationDTO findById(String negotiationId, boolean includeDetails)
   }
 
   @Transactional
-  public void enablePosts(String negotiationId) {
-    updatePostStatus(negotiationId, true);
+  public void enablePrivatePosts(String negotiationId) {
+    updatePostStatus(negotiationId, PostType.PRIVATE, true);
   }
 
   @Transactional
-  public void disablePosts(String negotiationId) {
-    updatePostStatus(negotiationId, false);
+  public void disablePrivatePosts(String negotiationId) {
+    updatePostStatus(negotiationId, PostType.PRIVATE, false);
   }
 
-  private void updatePostStatus(String negotiationId, boolean enabled) {
+  @Transactional
+  public void disablePublicPosts(String negotiationId) {
+    updatePostStatus(negotiationId, PostType.PUBLIC, false);
+  }
+
+  @Transactional
+  public void disableAllPosts(String negotiationId) {
+    updatePostStatus(negotiationId, null, false);
+  }
+
+  private void updatePostStatus(
+      String negotiationId, @Nullable PostType postType, boolean enabled) {
     Negotiation negotiation =
         negotiationRepository
             .findById(negotiationId)
             .orElseThrow(() -> new EntityNotFoundException(negotiationId));
-    negotiation.setPostsEnabled(enabled);
+
+    if (postType == null || postType.equals(PostType.PRIVATE)) {
+      negotiation.setPrivatePostsEnabled(enabled);
+    }
+    if (postType == null || postType.equals(PostType.PUBLIC)) {
+      negotiation.setPublicPostsEnabled(enabled);
+    }
```

Comment:
is this method really necessary? Just have a fetchNegotiaiton and then set the posts. That's 2 lines of code per method which I think is fine